test/core/run.py

```diff
@@ -41,7 +41,12 @@ class RunTests(unittest.TestCase):
   def _runCommand(self, command, logPath, expectedExitCode = 0):
     with open(logPath, 'w+') as out:
       exitCode = subprocess.call(command, shell=True, stdout=out, stderr=subprocess.STDOUT)
-      self.assertEqual(expectedExitCode, exitCode, "failed with exit code %i (expected %i) for %s" % (exitCode, expectedExitCode, command))
+      if exitCode != expectedExitCode:
```

Comment:
Instead of reimplementing assertEqual, why not put a a try around the with and cat logPath in its handler?